src/ElectronicRegulator.cpp

```diff
@@ -662,161 +196,129 @@ double compute_feedforward(double pressure_setpoint, double hp) {
 
 }
 
-void setup() {
-    //Start with valve line perpendicular to body (90 degrees)
-    Serial.begin(115200);
-    while(!Serial);
-    p_buff = new Buffer(BUFF_SIZE);
-
-    delay(500);
+void sendToAC(Comms::Packet packet) {
+    Serial1.write(packet.id);
+    Serial1.write(packet.len);
+    Serial1.write(packet.timestamp, 4);
+    Serial1.write(packet.checksum, 2);
+    Serial1.write(packet.data, packet.len);
+    Serial1.write('\n');
+}
 
-    #ifndef USE_DASHBOARD
-    waitConfirmation();
-    // move to close motors
-    Serial.println("Zeroing valve");
-    #endif
 
+//TODO fill out command methods
+void zero(Comms::Packet packet) {
     speed = -150;
     runMotor();
     delay(1000);
     speed = 0;
     runMotor();
     // zero encoder value (so encoder readings range from -x (open) to 0 (closed))
-    encoder.write(-20);
-
-    #ifndef USE_DASHBOARD
-    waitConfirmation();
-    #endif
-    //motorDirTest();
-    // ptTest();
-    delay(500);
-    servoTest();
-    // pressurize_tank();
-    
-    ptTest();
-
-    #ifndef USE_DASHBOARD
-    Serial.println("Next input will start servo loop, starting setpoint = "+String(pressure_setpoint));
-    waitConfirmation();
-    #endif
-
-    // pressurize_tank();
-    // long startAngle = 300;
-    // long endAngle = 1300;
-    // long thirdAngle = 300;
-    // long flowDuration = 3000; //time in ms for one way
-    // Serial.println("Starting angle sweep from "+String(startAngle)+" to "+String(endAngle)+" then back to " + String(thirdAngle) + " over "+String(2*flowDuration)+" ms...");
-    
-    // potTest();
-    servoTest();
-    
-    // angleSweep(startAngle, endAngle, flowDuration, 0, 200);
-    // angleSweep(endAngle, thirdAngle, flowDuration, 5000, 500);
-    // ptTest();
-    // exit(0);
-    t2 = micros();
-    start_time = micros();
-    pressure_errorInt = 0;
+    encoder.write(-20);}
+
+void setPressureSetpoint(Comms::Packet packet) {
+    pressure_setpoint = Comms::packetGetFloat(&packet, 0);
 }
 
-long lastPrint = 0;
-// Start in closed position, angle should be 0
+void regulation(Comms::Packet packet) {
```

Comment:
what is this for?
might wanna add comments for each of these commands.
Also, will there be a command for stopping the flow?